src/Integration/LocalServices/ErrorListInfoBarController.cs

```diff
@@ -114,7 +114,9 @@ public void Refresh()
         // possible use #if DEBUG instead.
         private void AssertOnUIThread()
         {
-            Debug.Assert(this.host.UIDispatcher.CheckAccess(), "The controller needs to run on the UI thread");
+            // Note: "this.host.UIDispatcher.CheckAccess()" does not return the correct result when 
+            // VS is shutting down. However, ThreadHelper.CheckAccess() does.
+            Debug.Assert(ThreadHelper.CheckAccess(), "The controller needs to run on the UI thread");
```

Comment:
This is an odd case - I debugged, and `host.UIDispatcher.CheckAccess()` was returning false on shutdown, even though the _ManagedThreadId_ was 1. 
Using `ThreadHelper.CheckAccess()` returns the expected result, even in the case of shutdown.